code/modules/clothing/chameleon.dm

```diff
@@ -0,0 +1,234 @@
+/datum/action/item_action/chameleon/change
+	name = "Chameleon Change"
+	var/list/chameleon_blacklist = list()
+	var/list/chameleon_list = list()
+	var/chameleon_type = null
+	var/chameleon_name = "Item"
+
+/datum/action/item_action/chameleon/change/proc/initialize_disguises()
+	if(button)
+		button.name = "Change [chameleon_name] Appearance"
+	chameleon_blacklist += target.type
+	var/list/temp_list = typesof(chameleon_type)
+	for(var/V in temp_list - (chameleon_blacklist))
+		chameleon_list += V
+
+/datum/action/item_action/chameleon/change/proc/update_look(mob/user)
+	var/list/item_names = list()
+	var/obj/item/picked_item
+	for(var/U in chameleon_list)
+		var/obj/item/I = U
+		item_names += initial(I.name)
+	var/picked_name
+	picked_name = input("Select [chameleon_name] to change it to", "Chameleon [chameleon_name]", picked_name) in item_names
+	if(!picked_name)
+		return
+	for(var/V in chameleon_list)
+		var/obj/item/I = V
+		if(initial(I.name) == picked_name)
+			picked_item = V
+	if(!picked_item)
+		return
+	if(iscarbon(user))
+		var/mob/living/C = user
```

Comment:
you're checking if it's a carbon, but then you typecast it as a living, which one is it?